src/DotNetEnv/LoadOptions.cs

```diff
@@ -4,47 +4,54 @@ namespace DotNetEnv
 {
     public class LoadOptions
     {
-        public static readonly LoadOptions DEFAULT = new LoadOptions();
-
-        public bool SetEnvVars { get; }
-        public bool ClobberExistingVars { get; }
-        public bool OnlyExactPath { get; }
+        public bool SetEnvVars { get; set; }
+        public bool ClobberExistingVars { get; set; }
+        public bool OnlyExactPath { get; set; }
```

Comment:
The reason I do things the way I did before is to explicitly avoid setters. Immutability is a Good Thing, so I aim for that wherever feasible. Obviously here it's not a huge thing, but introducing mutability actually begs a question: what if the values get changed after loading? These are questions with no clear answer, which immutability avoids completely. You create a thing once and you can never change it, so if you pass that as the arg, that's what getting built. Mutability makes that a muddy picture.

So, what do you think makes your changes better than the mutability concerns?

Reply:
More broadly, I just don't personally see the object syntax after the ctor being cleaner than a proper ctor that takes injected args. Can you explain your thinking here?

Reply:
The object syntax is independent from the ctor which has to be updated for every new Property added to the options.
And it allows setting the values in more flexible ways, depending on the use-case.

But if immutability is a must have, with newer language-syntax you have the possibility to make the setters init-only, which achieves exactly that.

In general I agree that immutability is a good thing, but my experiences so far showed, that it is a never-ending story to prevent users from doing silly things ;)
Of course, the "framework" (if we call the provided library a framework) should encourage "good" behavior, but imho we don't need to consider every silly thing, like changing options after passing them, which would be the problematic pattern here...
Short: I think it prevents syntactic possibilities just because some people have to be forced not to change options after passing them...

With the object-syntax (and non-immutability) you are generally more flexible in usage, for example serializing and deserializing the object (without additional glue-code).
Maybe this is never needed here of course, this is my general approach to prefer the flexibility and at the same time using objects as if they actually are immutable, which is most of the time a good thing.
As an example here: Standard-Settings-Objects in my projects are mutable too, because I let the `IConfiguration` [something like `configuration.GetSection<SettingsType>("sectionName")`] create them, which needs public setters (maybe init-only works too, not sure, but I guess public setters are needed there).

But after all I still agree with you, this is not a necessary thing and using the ctor is more or less equal.

Reply:
Writing this I see a specific useCase, which is not that uncommon I guess:
If you want to store your LoadOptions in `appsettings.json`, you are forced to have GlueCode for instantiation.
With the public Setters you can use something like `configuration.GetSection<LoadOptions>("DotEnvLoadOptions")`.

Reply:
At a minimum, remove the `set:` and if it all works still, that would make it readonly

Reply:
That said, if you can elaborate on the app settings use case, I'd be interested.

Reply:
I added another Test to show the appSettings-UseCase instead of explaining it.

Doing this I figured out that it is mandatory to have a parameterless ctor to do that, the ctor with optional parameters only is not enough to enable ConfigurationBinding.
==> I added the additional ctor, the const is introduced to assure the correct DefaultValues without changing too much

This kind of json-configuration is configured by default in most ProjectTemplates I think.
It is named AppSettings.json I think, but that's only about naming of course.

But with the default Configuration you can also use EnvironmentVariables or args (the Main-Method args) for example.
E.g. with EnvVars you need to add the following:
```
LoadOptions__SetEnvVars=false
LoadOptions__ClobberExistingVars=false
LoadOptions__OnlyExactPath=false
```

Reply:
I guess one can ask, if it is relevant or necessary to use the Configuration to configure something which adds additional configuration :D
I would say it is rather an edge case, thinking more about it. Especially because we add DotEnv to have easy configuration, so why would we use configuration to configure the configuration...
But the possibility to locally override the DotEnv-Configuration without the need to change (repo-)code can be a cool thing for teams (if the benefits outplay the additional complexity).

Because I generally favor the approach of (mutable) POCOs, it feels better for me to do it the same way I would do it for any other configuration-class, to be able to instantiate it via Configuration or any other Mapper (like "custom" json deserialize etc.).